### PR TITLE
[SNMP][LLDP][multi-asic]: Modify LLDP test case to support multi-asic

### PR DESCRIPTION
**Pull request by SuvarnaMeenakshi**
[SNMP][LLDP][multi-asic]: Modify LLDP test case to support multi-asic 
platform. Changes made to retrieve minigraph facts from all namespaces.

Signed-off-by: SuvarnaMeenakshi <sumeenak@microsoft.com>

### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### What is the motivation for this PR?
SNMP LLDP test case should be modified to support multi-asic platform.
#### How did you do it?
- Retrieve minigraph facts from all namespaces.

#### How did you verify/test it?
Verified on multi-asic VS testbed and single asic VS testbed.
Output from multi-asic msft_four_asic_vs hwsku :
pytest --testbed=vms-kvm-four-asic-t1-lag --inventory=veos_vtb --testbed_file=vtestbed.csv --host-pattern=vlab-08 --module-path=../ansible/library "snmp/test_snmp_lldp.py" --show-capture=no --disable_loganalyzer --skip_sanity -v
/usr/local/lib/python2.7/dist-packages/ansible/parsing/vault/__init__.py:44: CryptographyDeprecationWarning: Python 2 is no longer supported by the Python core team. Support for it is now deprecated in cryptography, and will be removed in a future release.
  from cryptography.exceptions import InvalidSignature
============================================================================== test session starts ===============================================================================
platform linux2 -- Python 2.7.17, pytest-4.6.5, py-1.9.0, pluggy-0.13.1 -- /usr/bin/python
cachedir: .pytest_cache
metadata: {'Python': '2.7.17', 'Platform': 'Linux-5.4.0-1031-azure-x86_64-with-Ubuntu-18.04-bionic', 'Packages': {'py': '1.9.0', 'pytest': '4.6.5', 'pluggy': '0.13.1'}, 'Plugins': {u'repeat': u'0.9.1', u'ansible': u'2.2.2', u'xdist': u'1.28.0', u'html': u'1.22.1', u'forked': u'1.3.0', u'metadata': u'1.10.0'}}
ansible: 2.8.12
rootdir: /data/sonic-mgmt/tests, inifile: pytest.ini
plugins: metadata-1.10.0, forked-1.3.0, html-1.22.1, repeat-0.9.1, xdist-1.28.0, ansible-2.2.2
collected 1 item                                                                                                                                                                 

snmp/test_snmp_lldp.py::test_snmp_lldp PASSED                                                                                                                              [100%]

================================================================================ warnings summary ================================================================================
/usr/local/lib/python2.7/dist-packages/_pytest/config/__init__.py:538
  /usr/local/lib/python2.7/dist-packages/_pytest/config/__init__.py:538: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: tests.common.dualtor
    self.import_plugin(import_spec)

-- Docs: https://docs.pytest.org/en/latest/warnings.html

Single asic testbed:
pytest --testbed=vms-kvm-t0 --inventory=veos_vtb --testbed_file=vtestbed.csv --host-pattern=vlab-01 --module-path=../ansible/library "snmp/test_snmp_lldp.py" --show-capture=no --disable_loganalyzer --skip_sanity -v
/usr/local/lib/python2.7/dist-packages/ansible/parsing/vault/__init__.py:44: CryptographyDeprecationWarning: Python 2 is no longer supported by the Python core team. Support for it is now deprecated in cryptography, and will be removed in the next release.
  from cryptography.exceptions import InvalidSignature
============================================================================================ test session starts =============================================================================================
platform linux2 -- Python 2.7.17, pytest-4.6.5, py-1.10.0, pluggy-0.13.1 -- /usr/bin/python
cachedir: .pytest_cache
metadata: {'Python': '2.7.17', 'Platform': 'Linux-5.4.0-66-generic-x86_64-with-Ubuntu-18.04-bionic', 'Packages': {'py': '1.10.0', 'pytest': '4.6.5', 'pluggy': '0.13.1'}, 'Plugins': {u'repeat': u'0.9.1', u'ansible': u'2.2.2', u'xdist': u'1.28.0', u'html': u'1.22.1', u'forked': u'1.3.0', u'metadata': u'1.11.0'}}
ansible: 2.8.12
rootdir: /data/sonic-mgmt/tests, inifile: pytest.ini
plugins: xdist-1.28.0, html-1.22.1, repeat-0.9.1, forked-1.3.0, metadata-1.11.0, ansible-2.2.2
collected 1 item                                                                                                                                                                                             

snmp/test_snmp_lldp.py::test_snmp_lldp PASSED                                                                                                                                                          [100%]

============================================================================================== warnings summary ==============================================================================================
/usr/local/lib/python2.7/dist-packages/_pytest/config/__init__.py:538
  /usr/local/lib/python2.7/dist-packages/_pytest/config/__init__.py:538: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: tests.common.dualtor
    self.import_plugin(import_spec)

-- Docs: https://docs.pytest.org/en/latest/warnings.html

### CODE REVIEW

File: tests/snmp/test_snmp_lldp.py [hunk 1]
```diff
@@ -34,7 +34,11 @@ def test_snmp_lldp(duthosts, rand_one_dut_hostname, localhost, creds, tbinfo):
     hostip = duthost.host.options['inventory_manager'].get_host(duthost.hostname).vars['ansible_host']
 
     snmp_facts = localhost.snmp_facts(host=hostip, version="v2c", community=creds["snmp_rocommunity"])['ansible_facts']
-    mg_facts   = duthost.get_extended_minigraph_facts(tbinfo)
+    mg_facts = {}
+    for asic_id in duthost.get_asic_ids():
```
Comment (smaheshm) [inline tests/snmp/test_snmp_lldp.py:38]: get_frontend_asic_ids()?

Comment (SuvarnaMeenakshi) [reply tests/snmp/test_snmp_lldp.py:38]: SNMP provides LLDP information for all ASICs and not just for front end asics.

File: tests/snmp/test_snmp_lldp.py [hunk 2]
```diff
@@ -80,8 +84,12 @@ def test_snmp_lldp(duthosts, rand_one_dut_hostname, localhost, creds, tbinfo):
     assert len(active_intf) >= len(minigraph_lldp_nei) * 0.8
 
     # skip neighbors that do not send chassis information via lldp
-    lldp_facts = duthost.lldp()['ansible_facts']
-    nei = [k for k, v in lldp_facts['lldp'].items() if k != 'eth0' and v['chassis'].has_key('mgmt-ip') ]
+    lldp_facts= {}
+    for asic_id in duthost.get_asic_ids():
```
Comment (smaheshm) [inline tests/snmp/test_snmp_lldp.py:88]: get_frontend_asic_ids()?

File: tests/snmp/test_snmp_lldp.py (hunk 1 above)
Comment (smaheshm) [reply tests/snmp/test_snmp_lldp.py:38]: oh ok. So you also test the backend ASIC as neighbors.

**Review by smaheshm — APPROVED** [review]

File: tests/snmp/test_snmp_lldp.py (hunk 1 above)
Comment (SuvarnaMeenakshi) [reply tests/snmp/test_snmp_lldp.py:38]: yes, that is right.

**Review by abdosi — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**